sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/With.scala

```diff
@@ -112,6 +112,22 @@ object With {
     With(replaced(commonExprRefs), commonExprDefs)
   }
 
+  /**
+   * Helper function to create a [[With]] statement when push down filter.
+   * @param expr original expression
+   * @param replaceMap Replaced attributes and common expressions
+   */
+  def apply(expr: Expression, replaceMap: Map[Attribute, Expression]): With = {
+    val commonExprDefsMap = replaceMap.map(m => m._1 -> CommonExpressionDef(m._2))
```

Comment:
why do we create a map if we never look up from it?

Reply:
Convenient to generate commonExprRefsMap.